vm/ubpf_vm.c

```diff
@@ -160,10 +160,14 @@ ubpf_exec(const struct ubpf_vm *vm, void *mem, size_t mem_len, uint64_t* bpf_ret
     reg[2] = (uint64_t)mem_len;
     reg[10] = (uintptr_t)stack + sizeof(stack);
 
+    inst_cnt->inst_vm = 0;
```

Comment:
You should check if inst_cnt is not NULL. ubpf_exec() is exported by inc/ubpf.h, so an external user might send NULL.

In fact, I'd suggest that instead of using the `enable` boolean on the struct, I'd just make this function accept either a NULL or a struct ubpf_inst_cnt. This should make it easier for callers that don't care about instruction counting.